samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -37,6 +37,7 @@ object StreamConfig {
   val BOOTSTRAP =               SAMZA_PROPERTY + "bootstrap"
   val PRIORITY =                SAMZA_PROPERTY + "priority"
   val IS_INTERMEDIATE =         SAMZA_PROPERTY + "intermediate"
+  val IS_BOUNDED =              SAMZA_PROPERTY + "bounded"
```

Comment:
This might be ok for now, but it feels like we should determine whether a stream is bounded from metadata, or perhaps its type. (e.g. I can't think of how an unbounded HDFS file could exist)

Reply:
I assume that's the job of StreamDescriptor :). It should generate the StreamSpec correctly.

This is also used in Beam where I translate BoundedSource to a bounded StreamSpec.

Reply:
I should have been more specific. We absolutely need the notion of whether a stream is bounded or not. 

Rather, my concern was about adding such a property to config.